docfiles/pxtweb/cookieCompliance.ts

```diff
@@ -0,0 +1,217 @@
+/// <reference path="../../localtypings/mscc.d.ts" />
+
+namespace pxt {
+    interface CookieBannerInfo {
+        /* Does the banner need to be shown? */
+        IsConsentRequired: boolean;
+
+        /* Name of the cookie, usually MSCC */
+        CookieName: string;
+
+        /* HTML for the banner to be embedded into the page */
+        Markup: string;
+
+        /* Scripts to be loaded in the page for the banner */
+        Js: string[];
+
+        /* CSS files to be loaded in the page for the banner*/
+        Css: string[];
+
+        /* The minimum date for which consent is considered valid (any consent given before this date does not count) */
+        MinimumConsentDate: string;
+
+        /* Error message from the server, if present */
+        Error?: string;
+    }
+
+    interface HttpResponse {
+        status: number;
+        body: string;
+    }
+
+    interface Callback<T> {
+        (err?: any, res?: T): void;
+    }
+
+    export function initAnalyticsAsync() {
+        getCookieBannerAsync(detectLocale(), document.domain, (bannerErr, info) => {
+            if (bannerErr || info.Error) {
+                // Start app insights, just don't drop any cookies
+                initializeAppInsights(false);
+                return;
+            }
+
+            // Clear the cookies if the consent is too old, mscc won't do it automatically
+            if (isConsentExpired(info.CookieName, info.MinimumConsentDate)) {
+                const definitelyThePast = new Date(0).toUTCString();
+                document.cookie = `ai_user=; expires=${definitelyThePast}`;
+                document.cookie = `ai_session=; expires=${definitelyThePast}`;
+                document.cookie = `${info.CookieName}=0; expires=${definitelyThePast}`;
+            }
+
+            let bannerDiv = document.getElementById("cookiebanner");
+            if (!bannerDiv) {
+                bannerDiv = document.createElement("div");
+                document.body.insertBefore(bannerDiv, document.body.firstChild);
+            }
+
+            // The markup is trusted because it's from our backend, so it shouldn't need to be scrubbed
+            bannerDiv.innerHTML = info.Markup;
+
+            if (info.Css && info.Css.length) {
+                info.Css.forEach(injectStylesheet)
+            }
+
+            all(info.Js || [], injectScriptAsync, msccError => {
```

Comment:
it practices there's only 1 javascript file (and there should never be more than one)

Reply:
The API returns an array so I support an array.